src/lib/server/models.ts

```diff
@@ -84,11 +73,65 @@ const modelConfig = z.object({
 
 const modelsRaw = z.array(modelConfig).parse(JSON5.parse(MODELS));
 
+async function getChatPromptRender(
+	m: z.infer<typeof modelConfig>
+): Promise<ReturnType<typeof compileTemplate<ChatTemplateInput>>> {
+	if (m.chatPromptTemplate) {
```

Comment:
shouldnt the order of fallback be the other way around? First, try to use transfomersjs, then try to use chatPromptTemplate

Reply:
The downside of doing it this way is that for example in prod, we specify the tokenizer for token counting but we actually want to override the chat template with our own template. 

I think in terms of specificity it makes sense to be `custom chat template > tokenizer "default" template` wdyt?

Reply:
yep sounds good. And if we want to change the order, let's do it in follow-up PR. Also, if we want to change the order, we would need to handle https://github.com/huggingface/chat-ui/pull/744#discussion_r1551321918 first

Reply:
yes agreed, overall we should push towards using tokenizers rather than chat prompt templates, but we will need to handle the edge cases before indeed 😁